pkg/tsdb/cloudwatch/time_series_query.go

```diff
@@ -98,6 +100,13 @@ func (e *cloudWatchExecutor) executeTimeSeriesQuery(ctx context.Context, req *ba
 	}
 
 	if err := eg.Wait(); err != nil {
+		var awsErr awserr.RequestFailure
```

Comment:
This error is coming from an error happening inside single query goroutine. To me the errors happening inside query goroutine should be put into dataframe, like we do it for sql data sources as an example https://github.com/grafana/grafana/blob/738a3a2d89bfc8474cf941a34d8da409b15a85aa/pkg/tsdb/sqleng/sql_engine.go#L209. This will allow data to be rendered if all queries except one fails and a 400 status will be returned. 

Probably need to check with @grafana/cloud-datasources if my suggestion doesn't work for some reason.

Reply:
+1 to encode the error within a frame if that's the case.